src/utils/storage.ts

```diff
@@ -250,6 +250,60 @@ export async function processHistory(storage: IStorage, { rewardLock }: { reward
   await updateWalletMetadataFromProcessedTxData(storage, { maxIndexUsed, tokens });
 }
 
+/**
+ * Fetch and save the data of the token set on the storage
+ * @param {IStorage} storage - Storage to save the tokens.
+ * @param {Set<string>} tokens - set of tokens to fetch and save.
+ * @returns {Promise<void>}
+ */
+async function updateTokensData(storage: IStorage, tokens: Set<string>): Promise<void> {
+  async function fetchTokenData(uid: string): Promise<{
+    success: true,
+    name: string,
+    symbol: string,
+  } | { success: false, message: string }> {
+    let retryCount = 0;
+
+    while (retryCount <= 5) {
+      try {
+        // Fetch and return the api response
+        let result: {
+          success: true;
+          name: string;
+          symbol: string;
+        } | { success: false, message: string } = await new Promise((resolve) => {
+          return walletApi.getGeneralTokenInfo(uid, resolve);
+        });
+        return result;
+      } catch (err: unknown) {
+        // Increase the retry counter and try again
+        retryCount += 1;
+        continue;
```

Comment:
I think we should have a way to log what error is happening in the fullnode, to help debugging.

But I know that we don't have a good way to do this from the lib currently, and putting a `console.log` in it is not the best option.

So I'm letting this comment here just to increase awareness for this problem of not having a good way to log things from the wallet-lib.

Reply:
Done